OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -591,7 +591,7 @@ bool LoadAsset(IReadOnlyPackage package, string filename)
 
 		bool ShowSourceDropdown(DropDownButtonWidget dropdown)
 		{
-			var sourceName = new CachedTransform<IReadOnlyPackage, string>(GetSourceDisplayName);
+			var sourceName = new CachedTransform<IReadOnlyPackage, string>(source => GetSourceDisplayName(source, dropdown));
```

Comment:
IMO it would be better to put truncating here and truncate to the label widgets length using `WidgetUtils.TruncateLabelToTooltip`. (this will require changing relevant in yaml `LabelWidget`s to `LabelWithTooltipWidget`s 

While in the other usage truncate to `DropDownButtonWidget` like you currently have (just locally)

Reply:
Actually, this line is not functioning at all as intended. This whole line should be moved to SetupItem function and the string made a variable instead of being gotten via CashedTransform